### DISCUSSION TITLE
Reverse ENS resolution calls should be batched, if multicall specified in the client

### DISCUSSION THREAD

**Discussion by apoorvlathey** [wevm/viem#1123] [Idea / Feature Request]
I had a long list of addresses that I wanted to display as ens names, but the `publicClient.getEnsName()` calls aren't getting batched right now, leading to getting rate-limited by the RPC even though the client has batch specified with wait as high as 1 sec.

**Comment by jxom** [comment]
Can you provide a minimal reproduction?

**Reply by apoorvlathey** [reply]
Here I have a simple React App for fetching ENS for 10 addresses: https://codesandbox.io/s/objective-thompson-jzxfc7?file=/src/App.js
I would expect the RPC calls to be batched and go via multicall, but on reviewing the networks tab, I see 10 individual RPC requests being made for each of these calls.
![image](https://github.com/wagmi-dev/viem/assets/20395316/49b1c0ff-e42a-4143-9046-c73d7fae0c97)

**Reply by scottrepreneur** [reply]
Did this have an updated solution? Looks like others are [working around this also](https://www.loopcrypto.xyz/blog/how-we-used-viem-to-support-ens-names-in-loops-company-dashboard).

- Maybe covered by #3440

**Reply by apoorvlathey** [reply]
Thanks for sharing the workaround!

**Comment by BitWonka** [comment]
Hey @apoorvlathey 
there's a solution with https://github.com/ensdomains/ensjs-v3/ 

but yes i'd love to see this also in viem, i dont want to have to install another library just for ENS name lookup

**Reply by scottrepreneur** [reply]
Took a minute to find the [batching calls](https://github.com/ensdomains/ensjs/blob/main/docs/basics/batching-calls.md) docs.
